### PR TITLE
Forward compatibility for tippy

### PR DESCRIPTION
**Pull request by NotMyFault**
https://github.com/jenkinsci/jenkins/pull/6408

### CODE REVIEW

**Comment by NotMyFault** [comment]
Could someone take a look who has more experience with the ATH pipeline? I don't really see why it would fail, if I fire it up locally it starts well

**Comment by timja** [comment]
Started with https://github.com/jenkinsci/acceptance-test-harness/pull/760 and running the master branch

**Comment by timja** [comment]
> Started with #760 and running the master branch

Master seems to work, PRs are broken

**Comment by timja** [comment]
cc @jglick / @Vlatombe ref https://github.com/jenkinsci/parallel-test-executor-plugin/releases/tag/316.v1324c7927cc8

**Comment by timja** [comment]
It seems to be the `estimateTestsFromFiles` feature.

This can be trivially reproduced by creating a multibranch pipeline pointing at https://github.com/jenkinsci/acceptance-test-harness/

https://github.com/jenkinsci/acceptance-test-harness/blob/279a45837c1f1797ecae2ed98479e679d93ed359/Jenkinsfile#L22

**Comment by timja** [comment]
Reverting https://github.com/jenkinsci/parallel-test-executor-plugin/pull/169 fixes it

**Comment by timja** [comment]
> Could someone take a look who has more experience with the ATH pipeline? I don't really see why it would fail, if I fire it up locally it starts well

will be fixed by https://github.com/jenkinsci/parallel-test-executor-plugin/pull/186

I'll replay your build with a workaround

**Comment by timja** [comment]
Looks like a legit failure @janfaracik 

see `SshSlavesPluginTest#newAgent`

image

Clicking the add button does nothing when creating a credential on the new agent screen for ssh build agents

image

**Comment by janfaracik** [comment]
> Looks like a legit failure @janfaracik
> 
> see `SshSlavesPluginTest#newAgent`
> 
> it's trying to make this xhr request: image
> 
> Clicking the add button does nothing when creating a credential on the new agent screen for ssh build agents
> 
> image

Hmm not working on `master` either, investigating now

**Comment by timja** [comment]
@janfaracik Reverting that change doesn't seem to fix it.

@NotMyFault any ideas?

I'll keep poking at it a bit more but don't have much more time atm

**Comment by timja** [comment]
Seems this is a core bug introduced in 2.344

**Comment by timja** [comment]
bisected to https://github.com/jenkinsci/jenkins/pull/6464.

**Comment by timja** [comment]
Matrix auth plugin seems to be broken with this change, it doesn't add new rows, cc @janfaracik

**Comment by timja** [comment]
This branch fixes row adding but tooltips don't seem to be working:
https://github.com/jenkinsci/matrix-auth-plugin/compare/master...timja:tippy.js?expand=1

**Comment by janfaracik** [comment]
> This branch fixes row adding but tooltips don't seem to be working:
> https://github.com/jenkinsci/matrix-auth-plugin/compare/master...timja:tippy.js?expand=1

Opened a fix for that here - https://github.com/jenkinsci/matrix-auth-plugin/pull/120 - thanks!

**Comment by timja** [comment]
Doesn't seem to be a war to test here with non released plugins easily 😢

**Comment by timja** [comment]
`plugins.SshSlavesPluginTest` needs test changes

File: src/main/resources/ath-container/run.sh [hunk 1]
```diff
@@ -44,8 +44,8 @@ if [ ! -f $war ]; then
     mirrors=https://get.jenkins.io
     case "$war" in
         "latest")
-            war=jenkins-latest.war
-            url=$mirrors/war/latest/jenkins.war
+            #war=jenkins-latest.war
+            url=https://repo.jenkins-ci.org/snapshots/org/jenkins-ci/main/jenkins-war/2.346-SNAPSHOT/jenkins-war-2.346-20220502.145901-3.war
```
Comment (timja) [inline src/main/resources/ath-container/run.sh:48]:
```suggestion
            url=https://repo.jenkins-ci.org/snapshots/org/jenkins-ci/main/jenkins-war/2.356-SNAPSHOT/jenkins-war-2.356-20220622.075827-1.war
```

**Comment by janfaracik** [comment]
Opened a fix here https://github.com/jenkinsci/acceptance-test-harness/pull/795

File: src/main/resources/ath-container/run.sh [hunk 2]
```diff
@@ -44,7 +44,7 @@ if [ ! -f $war ]; then
     case "$war" in
         "latest")
             war=jenkins-latest.war
-            url=https://updates.jenkins.io/latest/jenkins.war
+            url=https://repo.jenkins-ci.org/snapshots/org/jenkins-ci/main/jenkins-war/2.356-SNAPSHOT/jenkins-war-2.356-20220622.075827-1.war
```
Comment (timja) [inline src/main/resources/ath-container/run.sh:47]:
```suggestion
            url=https://repo.jenkins-ci.org/snapshots/org/jenkins-ci/main/jenkins-war/2.377-SNAPSHOT/jenkins-war-2.377-20221103.085941-1.war
```

**Review by timja — APPROVED** [review]

**Pull request merged**